### PR TITLE
Maintain validators with voting power

### LINKED ISSUE #696: Maintain validators with voting power
In BandChain, a state is considered valid if it's signed by >2/3 of *voting power*. We need to update Solidity code to reflect that.

### LINKED ISSUE #738: Make bridge validator updatable from owner

### PR DESCRIPTION
**Pull request by taobun**
fixed #696 
fixed #738

### CODE REVIEW

File: bridges/evm/contracts/Bridge.sol [hunk 1]
```diff
@@ -1,37 +1,57 @@
...
-            address validator = _validators[idx];
+            ValidatorWithPower memory validator = _validators[idx];
             require(
-                !validators[validator],
+                validatorPowers[validator.addr] == 0,
                 "DUPLICATION_IN_INITIAL_VALIDATOR_SET"
             );
-            validators[validator] = true;
+            validatorPowers[validator.addr] = validator.power;
+            totalValidatorPower += validator.power;
```
Comment (sorawit) [inline bridges/evm/contracts/Bridge.sol]: Let's use safemath here and everywhere else

**Review by sorawit — CHANGES_REQUESTED** [review]

File: bridges/evm/contracts/Bridge.sol (hunk 1 above)
Comment (taobun) [reply bridges/evm/contracts/Bridge.sol]: Yes sir!

File: bridges/evm/contracts/Bridge.sol [hunk 2]
```diff
@@ -1,37 +1,61 @@
...
-contract Bridge is IBridge {
+contract Bridge is IBridge, Ownable {
     using BlockHeaderMerkleParts for BlockHeaderMerkleParts.Data;
     using IAVLMerklePath for IAVLMerklePath.Data;
     using TMSignature for TMSignature.Data;
+    using SafeMath for uint256;
 
     /// Mapping from block height to the hash of "zoracle" iAVL Merkle tree.
     mapping(uint256 => bytes32) public oracleStates;
     /// Mapping from an address to whether it's a validator.
```
Comment (sorawit) [inline bridges/evm/contracts/Bridge.sol]: This comment needs to change. to its voting power.

**Review by sorawit — APPROVED** [review]
LGTM.

**Review by sorawit — CHANGES_REQUESTED** [review]
Nevermind. please update the comment.

File: bridges/evm/contracts/Bridge.sol [hunk 3]
```diff
@@ -1,37 +1,61 @@
 pragma solidity 0.5.14;
 pragma experimental ABIEncoderV2;
 import {BlockHeaderMerkleParts} from "./BlockHeaderMerkleParts.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
```
Comment (sorawit) [inline bridges/evm/contracts/Bridge.sol]: Should we use the same import structure?

**Review by sorawit — CHANGES_REQUESTED** [review]
Nevermind. please update the comment.

**Review by sorawit — APPROVED** [review]
Looks good!

**Review by prin-r — APPROVED** [review]

**Pull request merged**